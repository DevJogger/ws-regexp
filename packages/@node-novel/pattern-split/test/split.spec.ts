import novelPatternSplit, { INovelPatternSplitOptions } from '../core';

describe(`novelPatternSplit`, () =>
{
	const tests = [
		[
			[`(女佣|女僕)`], [
			'(女佣|女僕)',
		],
		],
		[
			[`女佣|女僕`], [
			'女佣', '女僕',
		],
		],
		[
			[`(?:女佣|女僕)`], [
			'女佣', '女僕',
		],
		],
		[
			[new RegExp(`(女佣|女僕)`)], [
			'(女佣|女僕)',
		],
		],
		[
			[new RegExp(`女佣|女僕`)], [
			'女佣', '女僕',
		],
		],
		[
			[new RegExp(`(?:女佣|女僕)`)], [
			'女佣', '女僕',
		],
		],
	] as [[string, ...any[]], string[]][];

	tests.forEach(function (testcase)
	{
		test(String(testcase[0][0]), () =>
		{

			let actual = novelPatternSplit(...testcase[0]);
			let expected = testcase[1];

			expect(actual).toStrictEqual(expected);
			//expect(actual).toBeInstanceOf(Date);
			expect(actual).toMatchSnapshot();

		});
	})

})


describe(`breakingMode: true`, () =>
{
	const _options: INovelPatternSplitOptions = {
		breakingMode: true,
	};

	let tests = [
		[
			[`(女佣|女僕)`, _options], [
			'女佣', '女僕'
		],
		],
		[
			[`女佣|女僕`, _options], [
			'女佣', '女僕'
		],
		],
		[
			[`(?:女佣|女僕)`, _options], [
			'女佣', '女僕'
		],
		],
		[
			[new RegExp(`(女佣|女僕)`), _options], [
			'女佣', '女僕'
		],
		],
		[
			[new RegExp(`女佣|女僕`), _options], [
			'女佣', '女僕'
		],
		],
		[
			[new RegExp(`(?:女佣|女僕)`), _options], [
			'女佣', '女僕'
		],
		],
		[
			[/(?<![ァ-ヴーｱ-ﾝﾞｰ])(ソフィア|索菲亞)(?![ァ-ヴーｱ-ﾝﾞｰ])/, _options], [
			'ソフィア', '索菲亞'
		],
		],
		[
			[new RegExp(`(?<![ァ-ヴーｱ-ﾝﾞｰ])(震夜)(?![ァ-ヴーｱ-ﾝﾞｰ])`), _options], [
			'震夜'
		],
		],
	] as [[string, ...any[]], string[]][];

	tests.forEach(function (testcase)
	{
		test(String(testcase[0][0]), () =>
		{

			let actual = novelPatternSplit(...testcase[0]);
			let expected = testcase[1];

			expect(actual).toStrictEqual(expected);
			//expect(actual).toBeInstanceOf(Date);
			expect(actual).toMatchSnapshot();

		});
	})

})
