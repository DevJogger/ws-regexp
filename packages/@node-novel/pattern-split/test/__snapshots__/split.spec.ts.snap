// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakingMode: true (?:女佣|女僕) 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`breakingMode: true (女佣|女僕) 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`breakingMode: true /(?:女佣|女僕)/ 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`breakingMode: true /(?<![ァ-ヴーｱ-ﾝﾞｰ])(ソフィア|索菲亞)(?![ァ-ヴーｱ-ﾝﾞｰ])/ 1`] = `
[
  "ソフィア",
  "索菲亞",
]
`;

exports[`breakingMode: true /(?<![ァ-ヴーｱ-ﾝﾞｰ])(震夜)(?![ァ-ヴーｱ-ﾝﾞｰ])/ 1`] = `
[
  "震夜",
]
`;

exports[`breakingMode: true /(女佣|女僕)/ 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`breakingMode: true /女佣|女僕/ 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`breakingMode: true 女佣|女僕 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`novelPatternSplit (?:女佣|女僕) 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`novelPatternSplit (女佣|女僕) 1`] = `
[
  "(女佣|女僕)",
]
`;

exports[`novelPatternSplit /(?:女佣|女僕)/ 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`novelPatternSplit /(女佣|女僕)/ 1`] = `
[
  "(女佣|女僕)",
]
`;

exports[`novelPatternSplit /女佣|女僕/ 1`] = `
[
  "女佣",
  "女僕",
]
`;

exports[`novelPatternSplit 女佣|女僕 1`] = `
[
  "女佣",
  "女僕",
]
`;
